### ISSUE TITLE
[Bug] mim download fails with ModuleNotFoundError: No module named 'mmengine'

### ISSUE THREAD

**Issue by Kaeseknacker** [open-mmlab/mmclassification#1343]
### Branch

master branch (0.24 or other 0.x version)

### Describe the bug

I installed mmclassification like this: https://mmclassification.readthedocs.io/en/latest/install.html

But "Verify the installation" fails.
I couldn't download a model with this command:
`mim download mmcls --config resnet50_8xb32_in1k --dest .`

I get the following error:
```
mim download mmcls --config resnet50_8xb32_in1k --dest .
/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/_distutils_hack/__init__.py:18: UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
  warnings.warn(
/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Traceback (most recent call last):
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mim/commands/download.py", line 105, in download
    from mmengine import Config
ModuleNotFoundError: No module named 'mmengine'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/spraul/miniconda3/envs/mmcls25/bin/mim", line 8, in <module>
    sys.exit(cli())
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mim/commands/download.py", line 57, in cli
    download(package, configs, dest_root, check_certificate)
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mim/commands/download.py", line 108, in download
    from mmcv import Config
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/__init__.py", line 6, in <module>
    from .fileio import *
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/fileio/__init__.py", line 2, in <module>
    from .file_client import BaseStorageBackend, FileClient
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/fileio/file_client.py", line 15, in <module>
    from mmcv.utils.misc import has_method
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/utils/__init__.py", line 41, in <module>
    from .env import collect_env
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/utils/env.py", line 13, in <module>
    from .parrots_wrapper import get_build_config
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/utils/parrots_wrapper.py", line 101, in <module>
    BuildExtension, CppExtension, CUDAExtension = _get_extension()
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/mmcv/utils/parrots_wrapper.py", line 71, in _get_extension
    from torch.utils.cpp_extension import (BuildExtension, CppExtension,
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/torch/utils/cpp_extension.py", line 8, in <module>
    import setuptools
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/setuptools/__init__.py", line 8, in <module>
    import _distutils_hack.override  # noqa: F401
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/_distutils_hack/override.py", line 1, in <module>
    __import__('_distutils_hack').do_override()
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/_distutils_hack/__init__.py", line 77, in do_override
    ensure_local_distutils()
  File "/home/spraul/miniconda3/envs/mmcls25/lib/python3.8/site-packages/_distutils_hack/__init__.py", line 64, in ensure_local_distutils
    assert '_distutils' in core.__file__, core.__file__
AssertionError: /home/spraul/miniconda3/envs/mmcls25/lib/python3.8/distutils/core.py
```

### Environment

```python
{'sys.platform': 'linux',
 'Python': '3.8.16 (default, Jan 17 2023, 23:13:24) [GCC 11.2.0]',
 'CUDA available': True,
 'GPU 0': 'NVIDIA GeForce RTX 2070 SUPER',
 'CUDA_HOME': None,
 'GCC': 'gcc (Debian 10.2.1-6) 10.2.1 20210110',
 'PyTorch': '1.12.1',
 'PyTorch compiling details': 'PyTorch built with:\n'
                              '  - GCC 9.3\n'
                              '  - C++ Version: 201402\n'
                              '  - Intel(R) oneAPI Math Kernel Library Version '
                              '2021.4-Product Build 20210904 for Intel(R) 64 '
                              'architecture applications\n'
                              '  - Intel(R) MKL-DNN v2.6.0 (Git Hash '
                              '52b5f107dd9cf10910aaa19cb47f3abf9b349815)\n'
                              '  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n'
                              '  - LAPACK is enabled (usually provided by '
                              'MKL)\n'
                              '  - NNPACK is enabled\n'
                              '  - CPU capability usage: AVX2\n'
                              '  - CUDA Runtime 11.3\n'
                              '  - NVCC architecture flags: '
                              '-gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37\n'
                              '  - CuDNN 8.3.2  (built against CUDA 11.5)\n'
                              '  - Magma 2.5.2\n'
                              '  - Build settings: BLAS_INFO=mkl, '
                              'BUILD_TYPE=Release, CUDA_VERSION=11.3, '
                              'CUDNN_VERSION=8.3.2, '
                              'CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, '
                              'CXX_FLAGS= -fabi-version=11 -Wno-deprecated '
                              '-fvisibility-inlines-hidden -DUSE_PTHREADPOOL '
                              '-fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM '
                              '-DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK '
                              '-DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE '
                              '-DEDGE_PROFILER_USE_KINETO -O2 -fPIC '
                              '-Wno-narrowing -Wall -Wextra '
                              '-Werror=return-type '
                              '-Wno-missing-field-initializers '
                              '-Wno-type-limits -Wno-array-bounds '
                              '-Wno-unknown-pragmas -Wno-unused-parameter '
                              '-Wno-unused-function -Wno-unused-result '
                              '-Wno-unused-local-typedefs -Wno-strict-overflow '
                              '-Wno-strict-aliasing '
                              '-Wno-error=deprecated-declarations '
                              '-Wno-stringop-overflow -Wno-psabi '
                              '-Wno-error=pedantic -Wno-error=redundant-decls '
                              '-Wno-error=old-style-cast '
                              '-fdiagnostics-color=always -faligned-new '
                              '-Wno-unused-but-set-variable '
                              '-Wno-maybe-uninitialized -fno-math-errno '
                              '-fno-trapping-math -Werror=format '
                              '-Werror=cast-function-type '
                              '-Wno-stringop-overflow, LAPACK_INFO=mkl, '
                              'PERF_WITH_AVX=1, PERF_WITH_AVX2=1, '
                              'PERF_WITH_AVX512=1, TORCH_VERSION=1.12.1, '
                              'USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, '
                              'USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, '
                              'USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, '
                              'USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, \n',
 'TorchVision': '0.13.1',
 'OpenCV': '4.7.0',
 'MMCV': '1.7.1',
 'MMCV Compiler': 'GCC 9.3',
 'MMCV CUDA Compiler': '11.3',
 'MMClassification': '0.25.0+3d4f80d'}
```

**Comment by Ezra-Yu** [comment]
```ModuleNotFoundError: No module named 'mmengine'```, this indicates that your installation was not successful.

try running the flowing command to install `mmengine`:
```mim install mmengine```

**Comment by mzr1996** [comment]
This should be an issue of `mim`. And we are trying to locate the problem.

**Comment by ice-tong** [comment]
Hi @Kaeseknacker @mzr1996, this is a known issue in setuptools: https://github.com/pypa/setuptools/issues/3621. And we will fix this problem in mim.

**Comment by Kaeseknacker** [comment]
Thanks for your replies. 
I tried it out a few weeks/months ago and it worked. So I just wait until it's fixed.

**Comment by zhouzaida** [comment]
> 

Hi @Kaeseknacker , we have fixed this issue and released a new version 0.3.6. You could upgrade it by `mim install -U openmim`.
